src/bufr/BufrParser/Query/QueryParser.cpp

```diff
@@ -0,0 +1,127 @@
+        working_str.erase(
+                std::remove(working_str.begin(), working_str.end(), ' '), working_str.end());
+
+        if (working_str.substr(0, 1) == "[") {
+            if (working_str.substr(working_str.length() - 1, 1) != "]") {
+                std::cerr << "Query Parser: multi query is lacking closing brackets." << std::endl;
+                exit(1);
+            }
+
+            working_str = working_str.substr(1, working_str.length() - 2);
+
+            std::vector<size_t> comma_positions;
+
+            size_t last_pos = 0;
+            while (working_str.find(',', last_pos) != std::string::npos)
+            {
+                auto comma_idx = working_str.find(',', last_pos);
+                comma_positions.push_back(comma_idx);
+                last_pos = comma_idx + 1;
+            }
+
+            last_pos = 0;
+            subqueries.reserve(comma_positions.size() + 1);
+            for (size_t commaIdx = 0; commaIdx < comma_positions.size() + 1; ++commaIdx)
+            {
+                if (commaIdx < comma_positions.size())
+                {
+                    subqueries.push_back(
+                            working_str.substr(last_pos, comma_positions[commaIdx] - last_pos));
+                    last_pos = comma_positions[commaIdx] + 1;
+                }
+                else
+                {
+                    subqueries.push_back(
+                            working_str.substr(last_pos, working_str.length() - last_pos));
+                }
+            }
+        }
+        else
+        {
+            subqueries.push_back(working_str);
+        }
+
+        return subqueries;
+    }
+
+
+    void QueryParser::splitQueryStr(const std::string& query,
+                                    std::string& subset,
+                                    std::vector<std::string>& mnemonics,
+                                    int& index) {
+        // Find positions of slashes
+        std::vector<size_t> slashPositions;
+        size_t slashIdx = 0;
+        size_t charIdx = 0;
+        for (charIdx = 0; charIdx < query.length(); ++charIdx) {
+            if (charIdx > 0 && query[charIdx] == '/') {
+                slashPositions.push_back(charIdx);
+                slashIdx++;
+            }
+        }
+
+        if (slashPositions.size() < 1) {
+            std::cerr << "Query Parser: Not a valid query string." << std::endl;
+            exit(1);
+        }
+
+        // Capture the subset string
+        subset = query.substr(0, slashPositions[0]);
+
+        std::vector<std::string> mnemonicStrings(slashPositions.size());
+
+        // Capture the sequence mnemonic strings
+        for (size_t mnemonicIdx = 0; mnemonicIdx < mnemonicStrings.size() - 1; ++mnemonicIdx)
+        {
+            mnemonicStrings[mnemonicIdx] =
+                    query.substr(slashPositions[mnemonicIdx] + 1,
+                                 slashPositions[mnemonicIdx + 1] - slashPositions[mnemonicIdx] - 1);
```

Comment:
What is trying to be done here?

Reply:
This line just splits the memonics in a query string into a list of strings. Substr first argument is position and the second position is the length of the string. Slash positions holds the positions of the "/" characters.